conformance/lang/expressions/literals/string_literal.balt

```diff
@@ -0,0 +1,372 @@
+
+    byte d = "c"; // @error string literal cannot be assigned to a variable of byte
+
+    float e = "44.5"; // @error string literal cannot be assigned to a variable of float
+}
+
+Test-Case: output
+Description: Test string literal in argument context.
+Labels: function-call-expr, function-defn, string
+
+public function main() {
+    printStringLiteral("Spec Tests"); // @output Spec Tests
+
+    printStringLiteral(")(*&^%$#><:~~!`"); // @output)(*&^%$#><:~~!`
+
+    printStringLiteral("12345\t\u{0039}"); // @output 12345   9
+
+    printStringLiteral("This code point first appeared in version 1.1 of the Unicode® Standard and belongs to the \"Basic Latin\" block which goes from 0x0 to 0x7F."); // @output This code point first appeared in version 1.1 of the Unicode® Standard and belongs to the "Basic Latin" block which goes from 0x0 to 0x7F.
+}
+
+function printStringLiteral(string str) {
+    io:println(str);
+}
+
+Test-Case: output
+Description: Test string literal in defaultable parameter context.
+Labels: function-call-expr, function-defn, defaultable-param, string
+
+public function main() {
+    printString("6564"); // @output 6564
+    printString(); // @output Default
+}
+
+function printString(string str = "Default") {
+    io:println(str);
+}
+
+Test-Case: output
+Description: Test string literal in method call expression context.
+Labels: method-call-expr, string
+
+public function main() {
+    io:println("some".startsWith("s")); // @output true
+
+    io:println("Hello World".substring(6)); // @output World
+
+    io:println("ABCD".toLowerAscii()); // @output abcd
+}
+
+Test-Case: output
+Description: Test string literal in type test expression context.
+Labels: is-expr, string
+
+public function main() {
+    io:println("Book" is string); // @output true
+
+    io:println("1234" !is string); // @output false
+
+    io:println("\t\n\r\\" is string); // @output true
+}
+
+Test-Case: output
+Description: Test string literal in return statement context.
+Labels: string, return-stmt
+
+public function main() {
+    io:println(getString1()); // @output Returned String
+
+    io:println(getString2()); // @output One
+                              // @output Two     245
+
+    io:println(getString3()); // @output RBG
+}
+
+function getString1() returns string {
+    return "Returned String";
+}
+
+function getString2() returns string {
+    return "One \nTwo\t245";
+}
+
+function getString3() returns string {
+    return "\u{0052}B\u{0047}";
+}
+
+Test-Case: output
+Description: Test string literal in match expression context.
+Labels: match-stmt, string
+
+public function main() {
+    match "Ballerina" {
+        var a => {
+            io:println(a); // @output Ballerina
+        }
+    }
+}
+
+Test-Case: output
+Description: Test string literal in structural constructor context.
+Labels: list-constructor-expr, mapping-constructor-expr, string, table-constructor-expr
```

Comment:
```suggestion
Labels: string, list-constructor-expr, mapping-constructor-expr, table-constructor-expr
```

Reply:
Shall we update other places too?

Reply:
We decided to use alphabetical order right?